iree/compiler/Conversion/LinalgToSPIRV/KernelDispatchUtils.cpp

```diff
@@ -350,11 +350,12 @@ struct TileWorkgroupSizePair {
   std::array<int64_t, 3> workgroupSize;
 };
 
-static LogicalResult getMaliSpecificConfig(linalg::ConvOp op,
+template <typename ConvOpTy>
+static LogicalResult getMaliSpecificConfig(ConvOpTy op,
```

Comment:
The configuration here assumes 2-d NHWC conv. So it cannot be extended to 1-d/3-d convolution. Could you split `getConvOpLaunchConfig` to two, one for 2-d conv, one for 1-D/3-D (which should not have Mali specific configuration)?

Reply:
How about adding a check if the rank equals to 4?

I think currently some passes only work on general conv op. The change is already too big, I would like to remove them in some follow up. WDYT?

Reply:
I think in the end, we will remove all linalg.conv op, and use conv_xd instead.

Reply:
A check to fail the match if it's not `linalg::ConvOp` or `linalg::ConvInputNHWCFilterHWCFOp` sounds good to me.

Reply:
Right. For now we need to recognize both of them. ;-P

Checking just the dimension might not work well; we need to check the op itself (as there are different layouts). This configuration only works with conv 2d nhwc.

Reply:
sounds good!